routes/api.rb

```diff
@@ -1,5 +1,76 @@
 class PatchApp
   hash_branch('api') do |r|
-    # /api branch handling
+    r.on 'offsets', Integer do |offset_id|
+      offset = Offset.dataset.find(offset_id).first
+
+      r.post 'proofs' do
+        params = typecast_params.convert!(symbolize: true) do |params|
+          params.float!('mass')
+          params.str!('serial_number')
+        end
+
+        proof = Proof.new({ mass_g: params[:mass], serial_number: params[:serial_number], offset_id: offset.id })
+
+        if proof.valid? && proof.save
+          response.status = 201
+          { id: proof.id, _links: { self: { href: "/api/proofs/#{proof.id}"} } }
+        else
+          response.status = 409
+          error_collection = proof.errors.map do |error|
+            description = error[1][0].split(':')[0]
+            remediation = error[1][0].split(':')[1]
+
+            { description: description, statusCode: 409, attribute: "proof.#{error[0].to_s}", remediation: remediation }
+          end
+
+          { _errors: error_collection }
+        end
+      end
+    end
+
+    r.get 'payments' do
+      payments = Order.join(:offsets, id: :offset_id)
+        .join(:projects, id: :project_id)
+        .select_all(:orders)
+        .select_append(:project_id, Sequel.as(:name, :project_name))
+        .where(captured: !(r.params['type'] == 'new'))
```

Comment:
can you explain this `!` syntax please?

Reply:
The bang syntax here would be inversion right? If 'new' was passed in the 'type' param it would turn this to false returning all un-captured orders.

Reply:
Ah okay, I'm used to seeing `!=` but that's just semantics, thanks!